core/src/main/java/org/apache/carbondata/core/metadata/schema/table/CarbonTable.java

```diff
@@ -132,11 +133,42 @@ private CarbonTable() {
     this.tablePrimitiveDimensionsMap = new HashMap<String, List<CarbonDimension>>();
   }
 
+  /**
+   * During creation of TableInfo from hivemetastore the DataMapSchemas and the columns
+   * DataTypes are not converted to the appropriate child classes.
+   *
+   * This method will cast the same to the appropriate classes
+   *
+   * @param tableInfo
+   */
+  private static void updateTableInfo(TableInfo tableInfo) {
+    List<DataMapSchema> dataMapSchemas = new ArrayList<>();
+    for (DataMapSchema dataMapSchema : tableInfo.getDataMapSchemaList()) {
```

Comment:
Similar code is present in `ThriftWrapperSchemaConverterImpl` also right? Is it possible to remove from there?